### PR TITLE
Allen/select timezone

### PR DESCRIPTION
**Pull request by algtan**
Added the ability to choose a timezone when searching for logs. The following was changed:
- Changed date format for `Timepicker` to show day and time
- Installed `react-timezone-select`
- Created new component `TimezonePicker` and added it to `Sidebar`

### CODE REVIEW

File: src/client/components/Sidebar.tsx [hunk 1]
```diff
@@ -53,11 +70,23 @@ class Sidebar extends PureComponent<SidebarProps, SidebarState> {
...
+          // Find offset between selected timezone and UTC in milliseconds
+          const selectedTimezoneOffset = timezone.offset * 60 * 60 * 1000
+          // Adjust 'start' date from local timezone to selected timezone
+          const adjustedStart = new Date(
+            start.getTime() - localTimezoneOffset - selectedTimezoneOffset
+          )
+          // Adjust 'end' date from local timezone to selected timezone
+          const adjustedEnd = new Date(
+            end.getTime() - localTimezoneOffset - selectedTimezoneOffset
+          )
```
Comment (thehobbit85) [inline src/client/components/Sidebar.tsx:85]: This part should be a utility function with tests.
This repo doesn't have tests yet.
You can copy the setup from the info server.

Comment (thehobbit85) [inline src/client/components/Sidebar.tsx]: This part should be a utility function with tests.
This repo doesn't have tests yet.
You can copy the setup from the info server.

File: src/client/components/TimezonePicker.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import React, { useState } from 'react'
```
Comment (thehobbit85) [inline src/client/components/TimezonePicker.tsx]: You should move this into the inputs file: https://github.com/EdgeApp/edge-logs-server/blob/master/src/client/components/Inputs.tsx

Then you would notice you can reuse the styles there.

File: src/client/components/Sidebar.tsx [hunk 3]
```diff
@@ -39,7 +49,14 @@ class Sidebar extends PureComponent<SidebarProps, SidebarState> {
...
       deviceInfo: '',
       userMessage: '',
-      userName: ''
+      userName: '',
+      timezone: {
+        value: 'GMT',
+        label: '(GMT+0:00) UTC',
+        abbrev: 'GMT',
+        offset: 0,
+        altName: 'GMT'
```
Comment (thehobbit85) [inline src/client/components/Sidebar.tsx]: This can be a `const` object at the top of the file.

**Review by thehobbit85 — CHANGES_REQUESTED** [review]
See comments

File: src/client/components/Sidebar.tsx (hunk 1 above)
Comment (algtan) [reply src/client/components/Sidebar.tsx]: As requested, this part has been broken out into a helper function (see fixup commit). As we discussed, PR to be resubmitted without unit tests. Mocha uses CommonJS modules, while react-timezone-select uses ES modules, causing problems with running the tests.

File: src/client/components/TimezonePicker.tsx (hunk 2 above)
Comment (algtan) [reply src/client/components/TimezonePicker.tsx]: As requested, this component was moved to the inputs file. TimezonePicker.tsx will be deleted in fixup commit.

File: src/client/components/Sidebar.tsx (hunk 3 above)
Comment (algtan) [reply src/client/components/Sidebar.tsx]: As requested, a `const` object was created with the initial timezone values (see fixup commit).

**Review by thehobbit85 — APPROVED** [review]

**Pull request merged**